File: backend/app/schemas/logs.py
"""日志查看 API DTO。"""

from pydantic import BaseModel, Field


class LogFileItem(BaseModel):
    name: str
    size: int
    modified_at: float


class LogFileListResponse(BaseModel):
    dir: str
    enabled: bool
    items: list[LogFileItem]


class LogContentResponse(BaseModel):
    file: str
    content: str
    offset: int
    reset: bool = False
    append: bool = False


class LogTailQuery(BaseModel):
    file: str = "app.log"
    tail_lines: int = Field(default=500, ge=1, le=5000, alias="tailLines")
    offset: int = Field(default=0, ge=0)

File: frontend/src/api/logs.ts
import { http } from './http'

export type LogFileItem = {
  name: string
  size: number
  modified_at: number
}

export type LogFileList = {
  dir: string
  enabled: boolean
  items: LogFileItem[]
}

export type LogContent = {
  file: string
  content: string
  offset: number
  reset: boolean
  append: boolean
}

export const fetchLogFiles = () => http.get<LogFileList>('/logs/files')

export const fetchLogContent = (params: {
  file?: string
  tailLines?: number
  offset?: number
}) =>
  http.get<LogContent>('/logs/content', {
    params: {
      file: params.file ?? 'app.log',
      tailLines: params.tailLines ?? 500,
      offset: params.offset ?? 0,
    },
  })
